### DISCUSSION TITLE
List Server Groups from all the pages.

### DISCUSSION THREAD

**Discussion by zyoann** [mitchos/pyZscaler#63] [Q&A] [answered]
Hello mitchos,

Thank you for your work on this and sharing.

We are trying to get the list of the Server Groups with list_groups() but we get only the first page out of 3.
How could we get them from others pages as well?

Thanks a lot.
Best,
zyoann

**Comment by mitchos** [comment, marked as answer]
Hi @zyoann 

We haven't wired the Iterator to handle pagination into the ZPA modules yet. 

I'm working on this in a feature branch at the moment so it's on its way. We were waiting for an update from one of our upstream dependencies to streamline this, which came a few days ago so we're ready to roll.

Can you let me know which endpoints you're currently using and need this for and I can get a fix out for you. I'm going to aim to have all the ZPA endpoints done tomorrow morning (GMT+10).

**Reply by zyoann** [reply]
Hello @mitchos 

Thanks for your feedback.

Currently we are looking for the ZPA Server groups - is that endpoints you are talking about?

**Reply by mitchos** [reply]
Yep, is that the only one right now?

**Reply by zyoann** [reply]
Yep the only one for now. 👍

**Reply by mitchos** [reply]
Hi @zyoann this should now be fixed in v0.12.0 (See #66)

Please update and let me know if you are having any issues. Docstrings haven't been updated yet so please note that if you want to change the page size, you'll need to pass `pagesize=<int>` as a kwarg in the list method.

Thanks for bringing this to our attention.

**Reply by zyoann** [reply]
Hello @mitchos - It's working perfectly.

Thank you for the fix and your time!

**Reply by mitchos** [reply]
My pleasure!

Best of luck with everything.
